test/timeout.js

```diff
@@ -0,0 +1,210 @@
+const test = require('brittle')
+const timers = require('../index.js')
+const { isAround, sleep } = require('./helpers/index.js')
+
+test('setTimeout', async function (t) {
+  t.plan(1)
+
+  const started = Date.now()
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 50), 'timers took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50)
+})
+
+test('interrupt setTimeout with CPU overhead', async function (t) {
+  t.plan(1)
+
+  const started = Date.now()
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 75), 'timers took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50)
+
+  while (Date.now() - started < 75) {} // eslint-disable-line no-empty
+
+  /* setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 125), 'native took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50) */
+})
+
+test('interrupt setTimeout with Atomics.wait', async function (t) {
+  t.plan(1)
+
+  const started = Date.now()
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 75), 'timers took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50)
+
+  sleep(75)
+
+  /* setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 125), 'native took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50) */
+})
+
+test('multiple setTimeout', async function (t) {
+  t.plan(4)
+
+  const started = Date.now()
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 20), '1st timer took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 20)
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 50), '2nd timer took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50)
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 20), '3rd timer took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 20)
+
+  timers.setTimeout(() => {
+    t.ok(isAround(Date.now() - started, 0), '4th timer took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 1)
+})
+
+test('clearTimeout', async function (t) {
+  t.plan(1)
+
+  const id = timers.setTimeout(() => t.fail('timeout should not be called'), 20)
+  setTimeout(() => timers.clearTimeout(id), 15)
+  setTimeout(() => t.pass(), 50)
+})
+
+test('clearTimeout twice', async function (t) {
+  t.plan(1)
+
+  const id = timers.setTimeout(() => t.fail('timeout should not be called'), 20)
+  setTimeout(() => {
+    timers.clearTimeout(id)
+    timers.clearTimeout(id)
+  }, 15)
```

Comment:
5ms lag is prop pushing it here for ci, to like 30ms or something

Reply:
True! Although the idea is to trigger it before, so I can move the order of the setTimeouts, so it won't matter?